crates/ruff_linter/src/message/sarif.rs

```diff
@@ -161,17 +213,60 @@ impl<'a> SarifResult<'a> {
         let start_location = message.expect_ruff_start_location();
         let end_location = message.expect_ruff_end_location();
         let path = normalize_path(&*message.expect_ruff_filename());
+        let uri = url::Url::from_file_path(&path)
+            .map_err(|()| anyhow::anyhow!("Failed to convert path to URL: {}", path.display()))?
+            .to_string();
+        let fixes = message
+            .fix()
+            .map(|fix| {
+                let fix_description = message
+                    .first_help_text()
+                    .map(std::string::ToString::to_string);
+
+                let replacements: Vec<SarifReplacement> = fix
+                    .edits()
+                    .iter()
+                    .map(|edit| SarifReplacement {
+                        deleted_region: SarifRegion {
+                            start_line: start_location.line,
+                            start_column: start_location.column,
+                            end_line: end_location.line,
+                            end_column: end_location.column,
+                        },
+                        inserted_content: InsertedContent {
+                            inserted_content: edit.content().unwrap_or("").to_string(),
+                        },
+                    })
+                    .collect();
+
+                let mut artifact_changes = Vec::new();
+                if !replacements.is_empty() {
+                    artifact_changes.push(SarifArtifactChange {
+                        uri: SarifArtifactLocation { uri: uri.clone() },
+                        replacements,
+                    });
+                }
```

Comment:
A fix is guaranteed (statically) to always have at least one edit
```suggestion
                
                artifact_changes.push(SarifArtifactChange {
                    uri: SarifArtifactLocation { uri: uri.clone() },
                    replacements,
                });
                
```